src/main/java/ru/testtask/service/FileService.java

```diff
@@ -0,0 +1,288 @@
+            } catch (Exception e) {
+                i++;
+                newOriginalFileName = buildOriginalFilenameWhileCopy(oldName, i);
+                log.warn("Trying...");
+            }
+        }
+    }
+
+    private String buildOriginalFilenameWhileCopy(String filename, int index){
+        return FilenameUtils.removeExtension(filename)
+                + " (" + index + ")." + FilenameUtils.getExtension(filename);
+    }
+
+    public Optional<FileData> findFileById(String id){
+        return fileDataRepo.findById(id);
+    }
+
+    public File uploadFileFromUrl(URL url) {
+        long fileMaxSizeByte = fileMaxSizeMb * 1048576;
+
+        HttpURLConnection conn = null;
+        try {
+            conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("HEAD");
+
+            if (conn.getContentLengthLong() < fileMaxSizeByte) {
+
+                File tempFile = File.createTempFile("tmp", null,
+                        new File(temporaryDir.toString()));
+                InputStream is = url.openStream();
+                OutputStream fos = new FileOutputStream(tempFile);
+                IOUtils.copy(is, fos);
+                fos.close();
+                return tempFile;
+            }
+            else {
+                throw new FileIsToLargeException(String.format("The file with URL %s is too large!", url.toString()));
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (conn != null) {
+                conn.disconnect();
+            }
+        }
+    }
+
+    public void getFileContent(String id, HttpServletResponse response) throws FileNotFoundException {
+        FileData fileData;
+
+        try {
+            fileData = getFileById(id);
+            InputStream is = new FileInputStream(String.valueOf(getFileAbsolutePath(fileData)));
+            IOUtils.copy(is, response.getOutputStream());
+            response.flushBuffer();
+        } catch (IOException ex) {
+            log.error("Error writing file to output stream. Filename was '{}'", id, ex);
+            throw new RuntimeException("IOError writing file to output stream");
+        }
+    }
+
+    public Path getFileAbsolutePath(FileData fileData){
+        Path path = Paths.get(storageRoot + File.separator + fileData.getDirectory() + File.separator + fileData.getFilename());
+        return path;
+    }
+
+    public Path getStorageRootPath(){
+        return storageRootPath;
+    }
+
+    public FileData getFileById(String id) throws FileNotFoundException {
+        Optional<FileData> optionalFileData = fileDataRepo.findById(id);
+        if (optionalFileData.isPresent()) {
+            return optionalFileData.get();
+        } else {
+            throw new FileNotFoundException(String.format("File with id %s does not found", id));
+        }
+    }
+
+    public String normalizeDirectory(String directory) throws InvalidPathException {
```

Comment:
Please use all of time when directory is tranfered from client, and please save normalized directory to db